### DISCUSSION TITLE
vector ignores last line of the file

### DISCUSSION THREAD

**Discussion by akunafin** [vectordotdev/vector#17813] [Q&A] [answered]
File source doesn't create an event for the last line in the file. 
In my log files last log line doesn't have a new line at the end, but I still would expect vector to generate an event for this line because it reaches an EOF. 

My vector source conf:
```
[sources.log_source]
  include = ["/log/clusterLogs/**/*.log"]
  type = "file"
  read_from = "beginning"
  remove_after_secs = 10
  ignore_checkpoints = true
  [sources.log_source.fingerprint]
    strategy = "device_and_inode"
```
Am I missing some configuration that will create an event for last line as well?

**Comment by jszwedko** [comment, marked as answer]
Hi @akunafin !

Unfortunately, without the newline, Vector cannot be sure that whatever is writing to the file has finished writing. Each line needs to end with a newline character to indicate it is a complete line. One thing we could do is have a timeout where, if there is no newline after `n` seconds, Vector will assume the line is complete. If this would be useful to you, could you open a feature request? I hope this helps!

**Reply by akunafin** [reply]
Hi @jszwedko !

Thanks for your reply. I found the way to always add a newline to the end of the file and now it works fine.
